File: cases/taint/scala/argument-position-negative/ArgumentPositionNegative.scala
package dataflowbench

object ArgumentPositionNegative {
  def dfb_source(): Int = { // DFB-SOURCE: argument-position-negative-input
    1
  }

  def chooseFirst(first: Int, second: Int): Int = { // DFB-WITNESS: argument-position-negative-first
    first
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: argument-position-negative-sink

  def run(): Unit = {
    val result = chooseFirst(0, dfb_source())
    dfb_sink(result)
  }
}

File: cases/taint/c/argument-position-negative/argument_position_negative.c
int dfb_source(void) { // DFB-SOURCE: argument-position-negative-input
    return 1;
}

int choose_first(int first, int second) { // DFB-WITNESS: argument-position-negative-first
    return first;
}

void dfb_sink(int value) {} // DFB-SINK: argument-position-negative-sink

void run(void) {
    int result = choose_first(0, dfb_source());
    dfb_sink(result);
}
